### PR TITLE
first multitenancy example

### PR DESCRIPTION
**Pull request by LucaLanziani**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Following a discussion in slack I'm adding a full example of the multi-tenancy implementation with EKS and cross account role

**Special notes for your reviewer**:

@sedefsavas better late than never ;) 

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by LucaLanziani** [comment]
It's 23:23 here and I obviously didn't check for typos, will do tomorrow

**Comment by richardcase** [comment]
Great to see this documentation, thanks @LucaLanziani :tada:

File: docs/book/src/SUMMARY_PREFIX.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
   - [Spot instances](./topics/spot-instances.md)
   - [Machine Pools](./topics/machinepools.md)
   - [Multi-tenancy](./topics/multitenancy.md)
+    - [Multi-tenancy example](./topics/full-multitenancy-implementation.md)
```
Comment (sedefsavas) [inline docs/book/src/SUMMARY_PREFIX.md:14]:
```suggestion
    - [Multi-tenancy in EKS-managed clusters](./topics/full-multitenancy-implementation.md)
```

File: docs/book/src/topics/full-multitenancy-implementation.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                 |               +--------------------+
+                 |               |                    |
+                 +---------------+->EKS - (Managed)   |
+                                 |                    |
+                                 +--------------------+
+```
+
+And specifically, we will only include:
+
+- AWS Account 0 (aka Manager account)
```
Comment (sedefsavas) [inline docs/book/src/topics/full-multitenancy-implementation.md:31]:
```suggestion
- AWS Account 0 (aka Manager account used by management cluster where cluster API controllers reside)
```

File: docs/book/src/topics/full-multitenancy-implementation.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+- AWS Account 1 (aka Managed account)
```
Comment (sedefsavas) [inline docs/book/src/topics/full-multitenancy-implementation.md:32]:
```suggestion
- AWS Account 1 (aka Managed account used for EKS-managed workload clusters)
```

File: docs/book/src/topics/full-multitenancy-implementation.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        - "arn:aws:iam::${AWS_MANAGER_ACCOUNT_ID}:oidc-provider/oidc.eks.${AWS_REGION}.amazonaws.com/id/${OIDC_PROVIDER_ID}"
+      Condition:
+        "ForAnyValue:StringEquals":
+          "oidc.eks.${AWS_REGION}.amazonaws.com/id/${OIDC_PROVIDER_ID}:sub":
+            - system:serviceaccount:capa-system:capa-controller-manager
+            - system:serviceaccount:capa-eks-control-plane-system:capa-eks-control-plane-controller-manager # Include if also using EKS
+EOL
+```
+
+Let's provision the Manager account with:
```
Comment (sedefsavas) [inline docs/book/src/topics/full-multitenancy-implementation.md:97]:
```suggestion
Let's provision the Manager role with:
```

File: docs/book/src/topics/full-multitenancy-implementation.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+### Install the CAPA provider in the manager cluster
+
+Here we install the Cluster API providers into the manager cluster and create a service account to use the `controllers.cluster-api-provider-aws.sigs.k8s.io` role for the Management Components.
+
+```bash
+export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-as-profile)
+export EKS=true
+export EXP_MACHINE_POOL=true
+export AWS_CONTROLLER_IAM_ROLE=arn:aws:iam::${AWS_MANAGER_ACCOUNT_ID}:role/controllers.cluster-api-provider-aws.sigs.k8s.io
+clusterctl init --kubeconfig manager.kubeconfig --infrastructure aws --target-namespace capi-providers
```
Comment (sedefsavas) [inline docs/book/src/topics/full-multitenancy-implementation.md:166]: @richardcase forgot if there was an issue with self-managed clusters for EKS. Would `clusterctl move` work for creating an EKS cluster then pivoting there?

File: docs/book/src/topics/full-multitenancy-implementation.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
+# Multitenancy setup with EKS and Service Account
+
+See [multitenancy](./multitenancy.md) for more
+details on enabling the functionality and the various options you can use.
+
+In this example, we are going to see how to create the following architecture with cluster API:
```
Comment (sedefsavas) [inline docs/book/src/topics/full-multitenancy-implementation.md:6]: A high level summary would be nice here like:

Creating a KinD cluster as bootstrap cluster with manager credentials, then creating an EKS-managed cluster and pivoting the management cluster to it. Then creating EKS-managed workload clusters in a different account using managed credentials.

File: docs/book/src/topics/full-multitenancy-implementation.md [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-as-profile)
+export EKS=true
+export EXP_MACHINE_POOL=true
+export AWS_CONTROLLER_IAM_ROLE=arn:aws:iam::${AWS_MANAGER_ACCOUNT_ID}:role/controllers.cluster-api-provider-aws.sigs.k8s.io
+clusterctl init --kubeconfig manager.kubeconfig --infrastructure aws --target-namespace capi-providers
+```
+
+## Managed cluster
+
+Time to build the managed cluster
```
Comment (sedefsavas) [inline docs/book/src/topics/full-multitenancy-implementation.md:171]:
```suggestion
Time to build the managed cluster for pivoting the bootstrap cluster.
```

**Comment by LucaLanziani** [comment]
Commits squashed

**Comment by sedefsavas** [comment]
Thanks @LucaLanziani 
/lgtm
/approve

**Pull request merged**